### PR TITLE
lte_link_control: enhance to support various networks

### PR DESCRIPTION
**Pull request by junqingzou**
Add below to the lte_lc_driver:
.PLMN lock, default false
.PDN connection authentication, default false
.Modem modes without GPS (LTE-M or NB-IoT only)
.Enable modem trace if configured in BSD lib
Correct PDP Context logging mistake
Enable LTE_M and GPS in prj.conf of application "assert_tracker"

Based on real need of testing in different networks.

Updated from [PR#1062](https://github.com/NordicPlayground/fw-nrfconnect-nrf/pull/1062) after rebasing master.

### CODE REVIEW

File: applications/asset_tracker/prj.conf [hunk 1]
```diff
@@ -24,6 +24,7 @@ CONFIG_MQTT_MAX_PACKET_LENGTH=2048
 # LTE link control
 CONFIG_POWER_OPTIMIZATION_ENABLE=n
 CONFIG_LTE_LINK_CONTROL=y
+CONFIG_LTE_NETWORK_MODE_LTE_M_GPS=y
```
Comment (jtguggedal) [inline applications/asset_tracker/prj.conf]: Thanks for this change! Could you also add it to `prj_nrf9160_pca20035ns.conf` for Thingy:91?

Comment (junqingzou) [reply applications/asset_tracker/prj.conf]: Thank you. Sorry to missed that and I'll add it.

Comment (DurandA) [reply applications/asset_tracker/prj.conf]: > Thanks for this change! Could you also add it to `prj_nrf9160_pca20035ns.conf` for Thingy:91?

Does this enable GPS when PSM is not available on Thingy:91?

Comment (junqingzou) [reply applications/asset_tracker/prj.conf]: This is only to configure modem mode to be LTE-M + GPS (which is the default mode before PR). The assert_tracker enable GPS by long press the button on Thingy:91, while PSM mode make sure GPS acquisition actually runs or not.

**Comment by lemrey** [comment]
Should we update the changelog for this?

**Comment by rlubos** [comment]
I'll update it.

**Pull request merged**